hooks/pre/30-upgrade.rb

```diff
@@ -80,6 +80,22 @@ def fix_katello_settings_file
   end
 end
 
+def mongo_mmapv1_check
+  custom_hiera = '/etc/foreman-installer/custom-hiera.yaml'
+  # check and see if we have a pulp_database already and if the Hiera file has already been modifed.
+  if File.file?('/var/lib/mongodb/pulp_database.0')
+    if File.foreach("#{custom_hiera}").grep(/mongodb::server::storage_engine:/).any?
```

Comment:
Would it make sense to just parse the YAML? Not sure if the ruby yaml parser maintains comments though.

Reply:
As long as we don't write the file, reading it as parsed yaml should be fine. This seems to be fairly clean though.